packages/fhir/src/Adaptor.js

```diff
@@ -38,202 +47,215 @@ export function execute(...operations) {
+      options
     );
-    const { baseUrl, apiPath } = state.configuration;
-
-    const url = `${baseUrl}/${apiPath}/${resolvedpath}`;
-
-    const headers = {
-      accept: 'application/fhir+json',
-      'Content-Type': 'application/fhir+json',
-    };
-
-    if (params?.headers) {
-      Object.assign(headers, params.headers);
-    }
 
-    const options = {
-      headers,
-      ...resolvedParams,
-      ...resolvedData,
-    };
-    return request(url, options, 'POST').then(({ response, data }) =>
-      handleResponse(response, data, state, callback)
+    const response = await util.request(
+      state.configuration,
+      'POST',
+      resolvedPath,
+      {
+        ...resolvedOptions,
+        body: resolvedData,
+      }
     );
+    return callback(util.prepareNextState(state, response));
   };
-}
 
 /**
- * Creates a resource in a destination system
+ * Send a HTTP GET request
  * @public
- * @example
- * create("Bundle", {
- *   entry: [
- *     {
- *       fullUrl: "", // Eg: Patient URL
- *       resource: {}, // Resource data
- *       search: {
- *         mode: "match",
- *       },
- *     },
- *   ],
- *   type: "collection",
- * });
  * @function
+ * @example <caption>Get Patient from FHIR</caption>
+ * get('Patient');
+ * @example <caption>Get Claim from FHIR with optional query</caption>
+ * get("Claim", { _include: "Claim:patient", _sort: "-_lastUpdated", _count: 200 })
  * @param {string} path - Path to resource
- * @param {object} data - data to create the new resource
+ * @param {object} params - Parameters to encode into the URL query
+ * @param {RequestOptions} options - Options to control the request
  * @param {function} callback - (Optional) callback function
  * @returns {Operation}
+ * @state {FHIRHttpState}
  */
-export function create(path, data, callback = s => s) {
-  return state => {
-    const [resolvedpath, resolvedData] = expandReferences(state, path, data);
+export const get =
+  (path, params = {}, options = {}, callback = s => s) =>
+  async state => {
+    const [resolvedPath, resolvedParams, resolvedOptions] = expandReferences(
+      state,
+      path,
+      params,
+      options
+    );
 
-    const { baseUrl, apiPath } = state.configuration;
+    const response = await util.request(
+      state.configuration,
+      'GET',
+      resolvedPath,
+      {
+        ...resolvedOptions,
+        query: resolvedParams,
+      }
+    );
 
-    const url = `${baseUrl}/${apiPath}/${resolvedpath}`;
+    return callback(util.prepareNextState(state, response));
+  };
 
-    const options = {
-      headers: {
-        accept: 'application/fhir+json',
-        'Content-Type': 'application/fhir+json',
-      },
-      ...resolvedData,
-    };
+/**
+ * Creates a resource in a destination system using a POST request
+ * @public
+ * @function
+ * @example
+ * create("Patient",{ ... })
+ * @param {string} resourceType - The resource type to create
+ * @param {object} resource - The resource to create
```

Comment:
The resource is an object, inside that object there are deferent keys depending on the resource your creating. For example

<details><summary>Here are sample resource key for Patient</summary>
<p>

```json
{
        "resourceType": "Patient",
        "meta": {
            "versionId": "1",
            "lastUpdated": "2019-07-01T13:30:55.864+00:00",
            "profile": [
                "https://www.ehealth.fgov.be/standards/fhir/StructureDefinition/be-patient"
            ]
        },
        "extension": [
            {
                "url": "http://hl7.org/fhir/StructureDefinition/patient-birthPlace",
                "valueAddress": {
                    "extension": [
                        {
                            "url": "http://hl7.org/fhir/StructureDefinition/language",
                            "valueCode": "nl"
                        }
                    ],
                    "city": "Namen",
                    "country": "BE"
                }
            },
            {
                "url": "http://hl7.org/fhir/StructureDefinition/patient-nationality",
                "extension": [
                    {
                        "url": "code",
                        "valueCodeableConcept": {
                            "coding": [
                                {
                                    "system": "https://www.ehealth.fgov.be/standards/fhir/CodeSystem/cd-fed-country",
                                    "code": "BE",
                                    "display": "Belgium"
                                }
                            ]
                        }
                    }
                ]
            }
        ],
        "identifier": [
            {
                "use": "official",
                "type": {
                    "coding": [
                        {
                            "system": "http://terminology.hl7.org/CodeSystem/v2-0203",
                            "code": "SB",
                            "display": "Social Beneficiary Identifier"
                        }
                    ]
                },
                "system": "https://www.ehealth.fgov.be/standards/fhir/NamingSystem/ssin",
                "value": "79121137740"
            },
            {
                "use": "usual",
                "type": {
                    "coding": [
                        {
                            "system": "http://terminology.hl7.org/CodeSystem/v2-0203",
                            "code": "MR",
                            "display": "Medical record number"
                        }
                    ]
                },
                "system": "https://www.goodhealthhospital.be/standards/fhir/NamingSystem/patientrecord",
                "value": "45XXP0PA-4"
            }
        ],
        "active": true,
        "name": [
            {
                "use": "official",
                "family": "La Paradisio",
                "given": [
                    "Josephine",
                    "Nessa"
                ]
            }
        ],
        "telecom": [
            {
                "system": "email",
                "value": "nessa.laparadisio@belgium.be"
            },
            {
                "system": "phone",
                "value": "+322476792979",
                "use": "mobile"
            },
            {
                "system": "phone",
                "value": "+3226718655",
                "use": "home"
            },
            {
                "system": "phone",
                "value": "+322476799",
                "use": "work"
            }
        ],
        "gender": "female",
        "birthDate": "1979-12-11",
        "address": [
            {
                "extension": [
                    {
                        "url": "http://hl7.org/fhir/StructureDefinition/language",
                        "valueCode": "nl"
                    }
                ],
                "use": "home",
                "type": "both",
                "text": "Sloordelle 42, 1160 Oudergem",
                "line": [
                    "Sloordelle 42"
                ],
                "_line": [
                    {
                        "extension": [
                            {
                                "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-streetName",
                                "valueString": "Sloordelle"
                            },
                            {
                                "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-houseNumber",
                                "valueString": "42"
                            }
                        ]
                    }
                ],
                "city": "Oudergem",
                "postalCode": "1160",
                "country": "BE"
            },
            {
                "extension": [
                    {
                        "url": "http://hl7.org/fhir/StructureDefinition/language",
                        "valueCode": "fr"
                    }
                ],
                "use": "home",
                "type": "both",
                "text": "42, Allee des Colzas, 1160 Auderghem",
                "line": [
                    "42, Allee des Colzas"
                ],
                "_line": [
                    {
                        "extension": [
                            {
                                "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-streetName",
                                "valueString": "Allee des Colzas"
                            },
                            {
                                "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-houseNumber",
                                "valueString": "42"
                            }
                        ]
                    }
                ],
                "city": "Auderghem",
                "postalCode": "1160",
                "country": "BE"
            },
            {
                "use": "work",
                "type": "both",
                "text": "377, Avenue Prince d'Orange, 1420 Braine-lʼAlleud",
                "line": [
                    "377, Avenue Prince d'Orange"
                ],
                "_line": [
                    {
                        "extension": [
                            {
                                "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-streetName",
                                "valueString": "Avenue Prince d'Orange"
                            },
                            {
                                "url": "http://hl7.org/fhir/StructureDefinition/iso21090-ADXP-houseNumber",
                                "valueString": "377"
                            }
                        ]
                    }
                ],
                "city": "Braine-lʼAlleud",
                "postalCode": "1420",
                "country": "BE"
            }
        ],
        "maritalStatus": {
            "coding": [
                {
                    "system": "http://terminology.hl7.org/CodeSystem/v3-MaritalStatus",
                    "code": "D",
                    "display": "Divorced"
                },
                {
                    "system": "https://www.ehealth.fgov.be/standards/fhir/CodeSystem/cd-civilstate",
                    "code": "41",
                    "display": "Divorced since 1/10/1994"
                }
            ]
        },
        "contact": [
            {
                "relationship": [
                    {
                        "coding": [
                            {
                                "system": "http://terminology.hl7.org/CodeSystem/v2-0131",
                                "code": "N",
                                "display": "Next-of-Kin"
                            },
                            {
                                "system": "https://www.ehealth.fgov.be/standards/fhir/CodeSystem/cd-contact-person",
                                "code": "mother"
                            }
                        ]
                    }
                ],
                "name": {
                    "family": "Vogels",
                    "given": [
                        "Leia"
                    ]
                },
                "telecom": [
                    {
                        "system": "phone",
                        "value": "+31201234567",
                        "use": "mobile"
                    }
                ]
            }
        ],
        "generalPractitioner": [
            {
                "reference": "Practitioner/01"
            }
        ]
    }
```

</p>
</details>